source/install/troubleshooting.rst

```diff
@@ -86,40 +84,193 @@ If email sign-in was turned off before the System Administrator switched sign-in
 
-First, ensure you have installed the `XML Security Library <https://www.aleksey.com/xmlsec/download.html>`__ on your Mattermost instance and that **it is available in your** PATH.
+``An error occurred while building Service Provider Metadata.``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This error indicates that the installation does not have an Enterprise license. The error message you receive will look similar to this:
+
+.. code-block:: sh
+     ERRO[2019-12-23T10:04:33.5074-07:00] An error occurred while building Service Provider Metadata.  caller="mlog/log.go:175" err_details="err=Your license does not support SAML authentication." err_where=GetSamlMetadata http_code=501 ip_addr="::1" method=GET path=/api/v4/saml/metadata request_id=fbtsbxzb33f67gn6yuy73asxjw user_id=
+
+To resolve the issue, install an Enterprise License and restart the process.
+
+
+``SAML 2.0 is not configured or supported on this server.``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The error indicates that the installation is using the Mattermost Team Edition. The error message you receive will look similar to this:
+
+.. code-block:: sh
+
+   ERRO[2019-12-23T10:12:56.5884001-07:00] An error occurred while building Service Provider Metadata.  caller="mlog/log.go:175" err_details="err=SAML 2.0 is not configured or supported on this server." err_where=GetSamlMetadata http_code=501 ip_addr="::1" method=GET path=/api/v4/saml/metadata request_id=1c7jrw3fzbggpe9rs83r5ge5fw user_id=
+
+To resolve the issue, install Enterprise Edition and restart the process.
+
+``An error occurred while initiating the request to the Identity Provider. Please contact your System Administrator``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This error message can have multiple causes. The log messages provide more information about the root cause and are provided below, along with a suggested fix.
+
+**Issue: Missing a Certificate File.**
+
+.. code-block:: sh
+
+   ERRO[2019-12-20T17:20:24.3999581-07:00] Identity Provider Public Certificate File was not found. Please contact your System Administrator.  caller="mlog/log.go:175" err_details= err_where=SamlInterfaceImpl.BuildRequest http_code=500 ip_addr="::1" method=GET path=/login/sso/saml request_id=tm9ywzxcbj88dypkhjgg8hideo user_id=
+
+Install the Identity Provider Certificate and restart the process.
+
+**Issue: Missing Service Provider Private Key**
+
+.. code-block:: sh
+
+   ERRO[2019-12-23T08:51:28.423397-07:00] An error occurred while configuring SAML Service Provider  caller="app/enterprise.go:154" error="saml-public.crt: cannot read: failed to get config file saml-private.key: failed to read file from /Users/sbishel/go/src/github.com/mattermost/mattermost-server/config/saml-private.key: open /Users/sbishel/go/src/github.com/mattermost/mattermost-server/config/saml-private.key: no such file or directory"
```

Comment:
I realized you just copied the error messages, I gave you. But can we change the paths to ".../mattermost-server/config/saml-private.key". The beginning part of the path will depend on their install anyway.

Reply:
Sure - I meant to ask you about that, so thanks for reminding me!